WordPress/gutenberg
# Implement Tabs in widget editor settings

## What?
Replace existing list-based tabs the new `Tabs` component in the site widget settings sidebar

## Why?
The current implementation was necessary because our legacy `TabPanel` component was not granular enough to be applied in this sidebar, due in large part to the internals of the the `ComplementaryArea` the sidebar uses.

## How?
Using the new, more granular sub-components of `Tabs`, we can render the appropriate parts of the UI via `ComplementaryArea`'s `slot`/`fill`, and pass the `Tabs` context to those components as needed.

This implementation is very similar to the one used previously in #55360. 

## Testing Instructions

Note: to test smaller/mobile viewports, make sure your viewport is at least 600px in width. Because it's technically a pinned item, [this sidebar is intentionally disabled on smaller screens](https://github.com/WordPress/gutenberg/blob/6189c2f9d2f6fd9eb8c2ad6dea4e11020b641ab3/packages/interface/src/components/pinned-items/style.scss#L4).

1. Activate a theme that supports widget areas, like Twenty Seventeen
2. Open **Appearance > Widgets**
3. In the sidebar test the **Widget Areas** and **Block** tabs.
5. Confirm that each tab shows the correct content
6. Click one of the displayed widget areas on the canvas. Confirm the **Widget Areas** tab updates to display the name of the currently active widget area
7. Select **Widget Areas** tab
8. In the canvas, click on a block and confirm that the **Block** tab is auto-selected with the correct content displayed
9. Test switching tabs while a block is selected, confirm it still functions as expected
10. Test opening and closing the sidebar, confirm there are no errors or crashes

### Testing Instructions for Keyboard
1. Launch the site editor
2. Press [Tab] until focus is applied to the currently active tab in the sidebar
4. Confirm arrow keys move you focus between tabs, but do not automatically select the focused tab
5. Confirm the [Tab] key leaves the `tablist` and focuses the next element instead (the close button)
6. Confirm that hitting [Tab] once more takes you to the first focusable element in the sidebar contents (specifically, you're making sure it doesn't focus the sidebar panel itself)
